File: Backend/API/Controllers/Models/CreateMessageInput.cs
namespace Backend.API.Controllers.Models
{
    public class CreateMessageInput
    {
        public string Text { get; set; }
        public DateTime SentTime { get; set; } 
        public string SenderId { get; set; }
        public string ReceiverId { get; set; }

        public CreateMessageInput(string text, DateTime sentTime, string senderId, string receiverId)
        {
            Text = text;
            SentTime = sentTime;
            SenderId = senderId;
            ReceiverId = receiverId;
        }
    }
}

File: frontend/src/models/CreateMessageInput.ts
export default class CreateMessageInput {
    Text: string;
    SentTime: Date;
    SenderId: string;
    ReceiverId: string;

    constructor(text: string, sentTime: Date, senderId: string, receiverId: string) {
        this.Text = text;
        this.SentTime = sentTime;
        this.SenderId = senderId;
        this.ReceiverId = receiverId;
    }
};